README.md

```diff
@@ -184,16 +184,91 @@ file inside your source code repository.
+    `MODULE_NAME:VARIABLE_NAME`, where `MODULE_NAME` is a full dotted path
+    of a module, and `VARIABLE_NAME` refers to a WSGI callable inside the
+    specified module.
+    Gunicorn will look for a WSGI callable named `application` if not specified.
+
+    If `APP_MODULE` is not provided, the `run` script will look for a `wsgi.py`
+    file in your project and use it if it exists.
+
+    If using `setup.py` for installing the application, the `MODULE_NAME` part
+    can be read from there. For example, see
+    [setup-test-app](https://github.com/openshift/sti-python/tree/master/3.3/test/setup-test-app).
 
 * **APP_CONFIG**
 
-    This variable indicates path to a module which contains [gunicorn configuration](http://docs.gunicorn.org/en/latest/configure.html).
+    Path to a valid Python file with
+    [gunicorn configuration](http://docs.gunicorn.org/en/latest/configure.html#configuration-file).
+
+* **DISABLE_COLLECTSTATIC**
+
+    Set it to a nonempty value to inhibit the execution of
+    'manage.py collectstatic' during the build. Only affects Django projects.
+
+* **DISABLE_MIGRATE**
+
+    Set it to a nonempty value to inhibit the execution of 'manage.py migrate'
+    when the produced image is run. Only affects Django projects.
+
+Source repository layout
+------------------------
+
+You need not change anything in your existing Python project's repository.
+However, if these files exist they affect the behavior of the build process:
+
+* **requirements.txt**
+
+  List of dependencies to be installed with `pip`. The format is documented
+  [here](https://pip.pypa.io/en/latest/user_guide.html#requirements-files).
+
+
+* **setup.py**
+
+  Configuration of various aspects of the project, including installation
+  dependencies, as documented
+  [here](https://packaging.python.org/en/latest/distributing.html#setup-py).
+  For most projects, it is sufficient to use `requirements.txt`.
+
+
+Run strategies
+--------------
+
+The Docker image produced by sti-python executes your project in one of these
+ways, in precedence order:
+
+* **Gunicorn**
+
+  The Gunicorn WSGI HTTP server is used to serve your application in case it is
+  installed. It can be installed by listing it either in the `requirements.txt`
+  file or in the `install_requires` section of the `setup.py` file.
+
+  If a file named `wsgi.py` is present in your repository, it will be used as
+  the entry point to your application. This can be overridden with the
+  environment variable `APP_MODULE`.
+  This file is present in Django projects by default.
+
+  If you have both Django and Gunicorn in your requirements, your Django project
+  will be automatically served with Gunicorn.
+
+* **Django development server**
+
+  If you have Django in your requirements, but don't have Gunicorn, then your
+  application will be served with Django's development web server. This is not,
+  however, a recommended way to serve your application in production.
```

Comment:
Make the last sentence about not recommended way of running up bold, to stress this. Still though I'm wondering if we want to write about this...